File: android-app/app/src/main/java/com/vektra/ehrms/ui/auth/OrganizationSignupScreen.kt
package com.vektra.ehrms.ui.auth

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.vektra.ehrms.data.repository.VektraRepository
import com.vektra.ehrms.ui.theme.*
import kotlinx.coroutines.launch

@Composable
fun OrganizationSignupScreen(
    onNavigateToLogin: () -> Unit
) {
    val repository = remember { VektraRepository() }
    val scope = rememberCoroutineScope()

    var companyName by remember { mutableStateOf("") }
    var adminName by remember { mutableStateOf("") }
    var email by remember { mutableStateOf("") }
    var password by remember { mutableStateOf("") }
    var isLoading by remember { mutableStateOf(false) }
    var errorMessage by remember { mutableStateOf<String?>(null) }
    var successMessage by remember { mutableStateOf<String?>(null) }

    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(Navy900)
            .padding(16.dp),
        contentAlignment = Alignment.Center
    ) {
        Card(
            modifier = Modifier
                .fillMaxWidth(0.95f)
                .verticalScroll(rememberScrollState()),
            shape = RoundedCornerShape(24.dp),
            colors = CardDefaults.cardColors(containerColor = Navy800)
        ) {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(24.dp),
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.spacedBy(16.dp)
            ) {
                Text(
                    text = "Register Organization",
                    fontSize = 24.sp,
                    fontWeight = FontWeight.Bold,
                    color = Emerald500
                )
                Text(
                    text = "Setup your multi-tenant Vektra HR workspace",
                    fontSize = 12.sp,
                    color = Slate400
                )

                errorMessage?.let { error ->
                    Surface(color = Rose600.copy(alpha = 0.2f), shape = RoundedCornerShape(12.dp), modifier = Modifier.fillMaxWidth()) {
                        Text(text = error, color = Rose600, modifier = Modifier.padding(12.dp), fontSize = 12.sp)
                    }
                }

                successMessage?.let { success ->
                    Surface(color = Emerald500.copy(alpha = 0.2f), shape = RoundedCornerShape(12.dp), modifier = Modifier.fillMaxWidth()) {
                        Text(text = success, color = Emerald500, modifier = Modifier.padding(12.dp), fontSize = 12.sp)
                    }
                }

                OutlinedTextField(
                    value = companyName,
                    onValueChange = { companyName = it },
                    label = { Text("Company / Organization Name", color = Slate400) },
                    leadingIcon = { Icon(Icons.Default.Business, contentDescription = null, tint = Emerald500) },
                    modifier = Modifier.fillMaxWidth(),
                    colors = OutlinedTextFieldDefaults.colors(focusedTextColor = Color.White, unfocusedTextColor = Color.White, focusedBorderColor = Emerald500, unfocusedBorderColor = Slate700)
                )

                OutlinedTextField(
                    value = adminName,
                    onValueChange = { adminName = it },
                    label = { Text("Admin Full Name", color = Slate400) },
                    leadingIcon = { Icon(Icons.Default.Person, contentDescription = null, tint = Emerald500) },
                    modifier = Modifier.fillMaxWidth(),
                    colors = OutlinedTextFieldDefaults.colors(focusedTextColor = Color.White, unfocusedTextColor = Color.White, focusedBorderColor = Emerald500, unfocusedBorderColor = Slate700)
                )

                OutlinedTextField(
                    value = email,
                    onValueChange = { email = it },
                    label = { Text("Admin Email", color = Slate400) },
                    leadingIcon = { Icon(Icons.Default.Email, contentDescription = null, tint = Emerald500) },
                    modifier = Modifier.fillMaxWidth(),
                    colors = OutlinedTextFieldDefaults.colors(focusedTextColor = Color.White, unfocusedTextColor = Color.White, focusedBorderColor = Emerald500, unfocusedBorderColor = Slate700)
                )

                OutlinedTextField(
                    value = password,
                    onValueChange = { password = it },
                    label = { Text("Password", color = Slate400) },
                    leadingIcon = { Icon(Icons.Default.Lock, contentDescription = null, tint = Emerald500) },
                    visualTransformation = PasswordVisualTransformation(),
                    modifier = Modifier.fillMaxWidth(),
                    colors = OutlinedTextFieldDefaults.colors(focusedTextColor = Color.White, unfocusedTextColor = Color.White, focusedBorderColor = Emerald500, unfocusedBorderColor = Slate700)
                )

                Button(
                    onClick = {
                        if (companyName.isBlank() || adminName.isBlank() || email.isBlank() || password.isBlank()) {
                            errorMessage = "All fields are required"
                            return@Button
                        }
                        isLoading = true
                        errorMessage = null
                        scope.launch {
                            val result = repository.organizationSignup(companyName.trim(), adminName.trim(), email.trim(), password)
                            isLoading = false
                            result.onSuccess { res ->
                                successMessage = res.message ?: "Organization registered successfully! You can now log in."
                            }.onFailure { err ->
                                errorMessage = err.message ?: "Registration failed."
                            }
                        }
                    },
                    modifier = Modifier.fillMaxWidth().height(50.dp),
                    shape = RoundedCornerShape(14.dp),
                    colors = ButtonDefaults.buttonColors(containerColor = Emerald500),
                    enabled = !isLoading
                ) {
                    if (isLoading) {
                        CircularProgressIndicator(color = Color.White, modifier = Modifier.size(24.dp))
                    } else {
                        Text("REGISTER WORKSPACE", fontWeight = FontWeight.Bold)
                    }
                }

                TextButton(onClick = onNavigateToLogin) {
                    Text("Already have an account? Sign In", color = Slate400, fontSize = 13.sp)
                }
            }
        }
    }
}

File: android/src/screens/auth/OrganizationSignupScreen.js
import React, { useState } from 'react';
import {
  View,
  Text,
  TextInput,
  TouchableOpacity,
  StyleSheet,
  ActivityIndicator,
  ScrollView,
} from 'react-native';
import api from '../../api/axios';
import { COLORS } from '../../theme/colors';

export default function OrganizationSignupScreen({ navigation }) {
  const [companyName, setCompanyName] = useState('');
  const [adminName, setAdminName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(null);
  const [success, setSuccess] = useState(null);

  const handleSignup = async () => {
    if (!companyName.trim() || !adminName.trim() || !email.trim() || !password.trim()) {
      setError('All fields are required.');
      return;
    }

    setLoading(true);
    setError(null);
    setSuccess(null);

    try {
      const res = await api.post('auth/organization-signup', {
        companyName: companyName.trim(),
        name: adminName.trim(),
        email: email.trim().toLowerCase(),
        password,
      });

      setLoading(false);
      setSuccess(res.data?.message || 'Organization registered successfully! You can now log in.');
    } catch (err) {
      setLoading(false);
      setError(err.response?.data?.message || 'Registration failed.');
    }
  };

  return (
    <ScrollView contentContainerStyle={styles.container}>
      <View style={styles.card}>
        <Text style={styles.title}>Register Organization</Text>
        <Text style={styles.subTitle}>Setup your multi-tenant Vektra HR workspace</Text>

        {error && (
          <View style={styles.errorBox}>
            <Text style={styles.errorText}>{error}</Text>
          </View>
        )}

        {success && (
          <View style={styles.successBox}>
            <Text style={styles.successText}>{success}</Text>
          </View>
        )}

        <TextInput
          style={styles.input}
          placeholder="Company / Organization Name"
          placeholderTextColor={COLORS.slate400}
          value={companyName}
          onChangeText={setCompanyName}
        />

        <TextInput
          style={styles.input}
          placeholder="Admin Full Name"
          placeholderTextColor={COLORS.slate400}
          value={adminName}
          onChangeText={setAdminName}
        />

        <TextInput
          style={styles.input}
          placeholder="Admin Email Address"
          placeholderTextColor={COLORS.slate400}
          value={email}
          onChangeText={setEmail}
          keyboardType="email-address"
          autoCapitalize="none"
        />

        <TextInput
          style={styles.input}
          placeholder="Password"
          placeholderTextColor={COLORS.slate400}
          value={password}
          onChangeText={setPassword}
          secureTextEntry
        />

        <TouchableOpacity style={styles.btn} onPress={handleSignup} disabled={loading}>
          {loading ? <ActivityIndicator color="#FFF" /> : <Text style={styles.btnText}>REGISTER WORKSPACE</Text>}
        </TouchableOpacity>

        <TouchableOpacity onPress={() => navigation.navigate('Login')} style={{ marginTop: 16 }}>
          <Text style={styles.linkText}>Already have an account? Sign In</Text>
        </TouchableOpacity>
      </View>
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: { flexGrow: 1, backgroundColor: COLORS.screenBg, justifyContent: 'center', padding: 20 },
  card: { backgroundColor: COLORS.cardBg, borderRadius: 24, padding: 24, alignItems: 'center' },
  title: { fontSize: 22, fontWeight: 'bold', color: COLORS.emerald500, marginBottom: 4 },
  subTitle: { fontSize: 12, color: COLORS.slate400, marginBottom: 20 },
  errorBox: { width: '100%', backgroundColor: 'rgba(225, 29, 72, 0.15)', borderRadius: 12, padding: 12, marginBottom: 16 },
  errorText: { color: COLORS.rose600, fontSize: 12, textAlign: 'center' },
  successBox: { width: '100%', backgroundColor: 'rgba(16, 185, 129, 0.15)', borderRadius: 12, padding: 12, marginBottom: 16 },
  successText: { color: COLORS.emerald500, fontSize: 12, textAlign: 'center' },
  input: { width: '100%', backgroundColor: COLORS.screenBg, borderRadius: 12, borderWidth: 1, borderColor: COLORS.slate700, paddingHorizontal: 14, height: 50, color: COLORS.textPrimary, marginBottom: 14 },
  btn: { width: '100%', height: 50, backgroundColor: COLORS.emerald500, borderRadius: 14, justifyContent: 'center', alignItems: 'center', marginTop: 10 },
  btnText: { color: '#FFF', fontWeight: 'bold', fontSize: 14 },
  linkText: { color: COLORS.slate400, fontSize: 13 },
});
